tutorials/tour/_posts/2017-02-13-self-types.md

```diff
@@ -0,0 +1,37 @@
+---
+layout: tutorial
+title: Self-type
+
+disqus: true
+
+tutorial: scala-tour
+categories: tour
+num: 25
+next-page: implicit-parameters
+previous-page: compound-types
+prerequisite-knowledge: nested-classes, mixin-class-composition
+---
+When writing extensible code, it can be useful to declare a self-type as a means of dependency injection.
+
+To use a self-type in a trait, write an identifier, the type of another trait to mix in, and a right fat arrow (e.g. `someIdentifier: SomeOtherTrait =>`).
```

Comment:
The syntax is actually weird. Maybe be up-front about it. `A self-type is a way to narrow the type of `this` or another identifier that aliases `this`. The syntax looks like normal function syntax but means something entirely different.` Darn, just noticed I introduced extra backticks.